From f4c0e627bd1b575047ad8fe0b9c3fc89c957fb42 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 2 Oct 2023 01:38:44 -0500
Subject: [PATCH 4/5] loader: Static library name related hacks

Re-adds the lib prefix and removes the so-ver suffixes

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 loader/CMakeLists.txt | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index d81d52917..548c7f379 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -391,14 +391,16 @@ if(WIN32)
     # when adding the suffix the import and runtime library names must be consistent
     # mingw: libvulkan-1.dll.a / vulkan-1.dll
     # msvc: vulkan-1.lib / vulkan-1.dll
-    set_target_properties(vulkan
-                          PROPERTIES
-                          OUTPUT_NAME ${API_TYPE}-1)
-    if(MINGW)
-        # generate the same DLL with mingw
+    if (NOT BUILD_STATIC_LOADER)
         set_target_properties(vulkan
                               PROPERTIES
-                              PREFIX "")
+                              OUTPUT_NAME ${API_TYPE}-1)
+        if(MINGW)
+            # generate the same DLL with mingw
+            set_target_properties(vulkan
+                                  PROPERTIES
+                                  PREFIX "")
+        endif()
     endif()
 
     if(MSVC AND ENABLE_WIN10_ONECORE)
@@ -563,7 +565,7 @@ if (PKG_CONFIG_FOUND)
         set(PRIVATE_LIBS "Libs.private: ${PRIVATE_LIBS}")
     endif()
 
-    if(WIN32)
+    if(WIN32 AND NOT BUILD_STATIC_LOADER)
         if(MINGW)
             set(VULKAN_LIB_SUFFIX "-1.dll")
         else()
-- 
2.49.0

